sros2/sros2/keystore/_enclave.py

```diff
@@ -83,39 +85,39 @@ def create_key(keystore_path, identity):
     enclave_element = policy_element.find('enclaves/enclave')
     enclave_element.attrib['path'] = identity
 
-    permissions_path = os.path.join(key_dir, 'permissions.xml')
+    permissions_path = key_dir.joinpath('permissions.xml')
     _permission.create_permission_file(permissions_path, _utilities.domain_id(), policy_element)
 
-    signed_permissions_path = os.path.join(key_dir, 'permissions.p7s')
-    keystore_permissions_ca_cert_path = os.path.join(
-        _keystore.get_keystore_public_dir(keystore_path), 'permissions_ca.cert.pem')
-    keystore_permissions_ca_key_path = os.path.join(
-        _keystore.get_keystore_private_dir(keystore_path), 'permissions_ca.key.pem')
+    signed_permissions_path = key_dir.joinpath('permissions.p7s')
+    keystore_permissions_ca_cert_path = _keystore.get_keystore_public_dir(
+        keystore_path).joinpath('permissions_ca.cert.pem')
+    keystore_permissions_ca_key_path = _keystore.get_keystore_private_dir(
+        keystore_path).joinpath('permissions_ca.key.pem')
     _utilities.create_smime_signed_file(
         keystore_permissions_ca_cert_path,
         keystore_permissions_ca_key_path,
         permissions_path,
         signed_permissions_path
     )
 
-    return True
 
+def get_enclaves(keystore_path: pathlib.Path) -> Set[str]:
+    if not _keystore.is_valid_keystore(keystore_path):
+        raise sros2.errors.InvalidKeystoreError(keystore_path)
 
-def list_keys(keystore_path):
     enclaves_path = _keystore.get_keystore_enclaves_dir(keystore_path)
-    if not os.path.isdir(keystore_path):
-        raise FileNotFoundError(
-            errno.ENOENT, os.strerror(errno.ENOENT), keystore_path)
-    if not os.path.isdir(enclaves_path):
-        return True
-    p = pathlib.Path(enclaves_path)
-    key_file_paths = sorted(p.glob('**/key.pem'))
-    for key_file_path in key_file_paths:
-        print('/{}'.format(key_file_path.parent.relative_to(enclaves_path).as_posix()))
-    return True
-
-
-def _is_key_name_valid(name):
+    enclaves: Set[str] = set()
+    if os.path.isdir(enclaves_path):
+        p = pathlib.Path(enclaves_path)
+        key_file_paths = p.glob('**/key.pem')
```

Comment:
if enclaves_path is now a pathlib.Path we can 
```suggestion
    if enclaves_path.is_dir():
        key_file_paths = enclaves_path.glob('**/key.pem')
```

Reply:
Thank you, fixed.